lib/ibm/cloud/sdk/vpc/base_mixins/has_child.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+# frozen_string_literal: true
+
+module IBM
+  module Cloud
+    module SDK
+      # Access a specific instance by either id or name depending on API.
+      module ChildMixin
+        # Get an instance of the collection object.
+        # @param id [String] ID of Name to search on depending on API.
+        # @return [] The instance object.
+        def instance(id)
+          @instance.new(self, id)
+        end
+
+        # Return the first_instance returned from a collection get.
+        def first_instance
+          result = params(limit: 1).all.first
```

Comment:
I wonder if the manageiq query_params gem would help here as it gives a chainable interface automatically.  That is, you'd get things like `instances.first`

Reply:
I can have a look.  I added this mostly to simplify testing the API. The params(limit: 1) is a chainable to set the Cloud API query parameters. In this case I'm mostly using it to save on data transference and VCR file size. After that is the 'all' enumerator and first is a standard method on the enumerator.

For the params methods. TheCloud API has common parameters between all classes.
https://github.com/jaredm-ibm/ibm-cloud-sdk-ruby/blob/9117591ca9bf75e931db3fe5ad1a2f735056df4b/lib/ibm/cloud/sdk/vpc/base_collection.rb#L34

However some APIs have additional params in child classes.
https://github.com/jaredm-ibm/ibm-cloud-sdk-ruby/blob/9117591ca9bf75e931db3fe5ad1a2f735056df4b/lib/ibm/cloud/sdk/vpc/instances.rb#L29

Do we gain anything from implementing another layer? Do we lose any flexibility to conform to the Cloud API?

I am toying with the idea of making BaseInstance a subclass of Hash. And having 'all' wrap the returned hashes automatically in an instance class. Which would do away with this method, and provide a cleaner interface. I still have analysis to do on the returned objects in the collection vs the ones returned when directly queried to see if this is a good idea. It might just introduce more complexity than it is worth.